aplib.net-demo/Assets/Scripts/Inventory/EquipmentInventory.cs

```diff
@@ -150,4 +120,55 @@ public void MovePrevious()
         int newIndex = (_currentEquipmentIndex - 1 + _equipmentList.Count) % _equipmentList.Count;
         SwitchEquipment(newIndex);
     }
+
+    /// <summary>
+    /// Tries to use the currently equipped equipment item.
+    /// </summary>
+    public void TryUseEquipment()
+    {
+        if (HasItems) CurrentEquipment.UseEquipment();
```

Comment:
You will still need to get the animator from somewhere, but this should do it
```suggestion
        if (HasItems) 
        {
            if (CurrentEquipment is Weapon weapon && weapon.CanFire())
                _playerAnimator.SetTrigger("PlayerAttack");

            CurrentEquipment.UseEquipment();
        }
```